src/functions/env.ts

```diff
@@ -240,6 +252,62 @@ export function hasUserEnvs({
   return findEnvfiles(functionsSource, projectId, projectAlias, isEmulator).length > 0;
 }
 
+/**
+ * Write new environment variables into a dotenv file.
+ *
+ * Identifies one and only one dotenv file to touch using the same rules as loadUserEnvs().
+ * It is an error to provide a key-value pair which is already in the file.
+ * Not actually implemented yet.
+ */
+export function writeUserEnvs(toWrite: Record<string, string>, envOpts: UserEnvsOpts) {
```

Comment:
I have a user experience question that i don't have a good answer for and would like to tag @taeold into here.

Should we actually be ever writing back to a project alias file or specifically a per-project file? E.g. if the alias is `dev` and we write back to that file, we're going to modify all dev projects for a codebase.